plugins/modules/foreman_hostgroup.py

```diff
@@ -0,0 +1,216 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_hostgroup
+short_description: Manage Foreman Hostgroups using Foreman API
+description:
+  - Create, Update and Delete Foreman Hostgroups using Foreman API
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+  - "apypie"
+options:
+  name:
+    description: Name of Hostgroup
+    required: true
+  description:
+    description: description of hostgroup
+    required: false
+  locations:
+    description: List of locations the hostgroup should be assigned to
+    required: false
+    default: None
+    type: list
+  organizations:
+    description: List of organizations the hostgroup should be assigned to
+    required: false
+    default: None
+    type: list
+  architecture:
+    description: Architecture name
+    required: False
+    default: None
+  domain:
+    description: Domain name
+    required: False
+    default: None
+  environment:
+    description: Environment name
+    required: False
+    default: None
+  medium:
+    description: Medium name
+    required: False
+    default: None
+  operatingsystem:
+    description: Operatingsystem name
+    required: False
+    default: None
+  parameters:
+    description: List of parameters and values
+    required: false
+    default: None
+  partition_table:
+    description: Partition table name
+    required: False
+    default: None
+  pxe_loader:
+    description: PXE Loader
+    required: False
+    default: None
+  realm:
+    description: Realm name
+    required: false
+    default: None
+  root_pass:
+    description: root password
+    required: false
+    default: None
+  force_update:
+    description: forces update of hostgroup, usefull when in need of password update
+    required: false
+    default: false
+  smart_proxy:
+    description: Smart Proxy name
+    required: False
+    default: None
+  subnet:
+    description: Subnet name
+    required: False
+    default: None
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: verify ssl connection when communicating with foreman
+    default: true
+    type: bool
+  state:
+    description: domain presence
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: "Create a Hostgroup"
+  foreman_hostgroup:
+    name: "new_hostgroup"
+    architecture: "architecture_name"
+    operatingsystem: "operatingsystem_name"
+    medium: "media_name"
+    ptable: "Partition_table_name"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
```

Comment:
I'd prefer if we'd not ship examples with `validate_certs=False` (or `validate_certs` at all, the user will often copy paste from the example, and not adding params that have a sane default sounds nice to me)